File: src/sustech-lambda/src/main/java/cn/edu/sustc/cse/ooad/sustechlambda/sustechlambda/security/SLUserDetails.java
package cn.edu.sustc.cse.ooad.sustechlambda.sustechlambda.security;

import org.springframework.security.core.GrantedAuthority;
import org.springframework.security.core.userdetails.UserDetails;

import java.util.Collection;

public class SLUserDetails implements UserDetails {

    private final String subject;

    public SLUserDetails(String subject) {
        this.subject = subject;
    }

    @Override
    public Collection<? extends GrantedAuthority> getAuthorities() {
        return null;
    }

    @Override
    public String getPassword() {
        return null;
    }

    @Override
    public String getUsername() {
        return this.subject;
    }

    @Override
    public boolean isAccountNonExpired() {
        return true;
    }

    @Override
    public boolean isAccountNonLocked() {
        return true;
    }

    @Override
    public boolean isCredentialsNonExpired() {
        return true;
    }

    @Override
    public boolean isEnabled() {
        return true;
    }
}

File: src/sustech-lambda/src/main/kotlin/cn/edu/sustc/cse/ooad/sustechlambda/sustechlambda/security/SLUserDetails.kt
package cn.edu.sustc.cse.ooad.sustechlambda.sustechlambda.security

import org.springframework.security.core.userdetails.UserDetails

class SLUserDetails(private val subject: String) : UserDetails {

    override fun getAuthorities() = null

    override fun getPassword() = null

    override fun getUsername() = this.subject

    override fun isAccountNonExpired() = true

    override fun isAccountNonLocked() = true

    override fun isCredentialsNonExpired() = true

    override fun isEnabled() = true
}
